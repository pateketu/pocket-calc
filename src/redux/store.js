import {createStore} from 'redux';
import calc from './calcReducer';

const store = createStore(calc);

export default store;
