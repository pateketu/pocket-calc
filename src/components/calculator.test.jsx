import * as React from 'react';
import {mount} from 'enzyme';
import Calculator from './calculator';
import {Provider} from 'react-redux';
import store from '../redux/store';
import toJson from 'enzyme-to-json';

describe('Calculator', ()=>{
	let wrapper; 
	beforeEach(()=>{		
		wrapper = mount(<Provider store={store}>
			<Calculator></Calculator>
		</Provider>);
	});	
	
	it('renders as expeced', ()=>{
		expect(toJson(wrapper, {noKey:true})).toMatchSnapshot();   
	});

	describe('calculation',()=>{
		/* 
			Clearing calculator after each test as tests are 
			simulating actual usage and in actual usage like this 
			test there will be only one copy of the state (redux store)
		*/
		afterEach(()=>{
			wrapper.find('button[children="C"]').simulate('click');
		});
	
		it('click --> 1 --> 5 -- > + --> 5 --> =, updates display with expected calculated value', ()=>{
			const expectedResult = '20';
			
			wrapper.find('button[children="1"]').simulate('click');
			wrapper.find('button[children="5"]').simulate('click');
			wrapper.find('button[children="+"]').simulate('click');
			wrapper.find('button[children="5"]').simulate('click');
			wrapper.find('button[children="="]').simulate('click');

			expect(wrapper.find('.display').text()).toBe(expectedResult);
		});

		it('click --> 0.1 -- > + --> 0.2 --> =, updates display with expected calculated value', ()=>{
			const expectedResult = '0.3';

			wrapper.find('button[children="0"]').simulate('click');
			wrapper.find('button[children="."]').simulate('click');
			wrapper.find('button[children="1"]').simulate('click');
			wrapper.find('button[children="+"]').simulate('click');
			wrapper.find('button[children="0"]').simulate('click');			
			wrapper.find('button[children="."]').simulate('click');
			wrapper.find('button[children="2"]').simulate('click');
			wrapper.find('button[children="="]').simulate('click');
			
			expect(wrapper.find('.display').text()).toBe(expectedResult);
		});

		it('click --> 5  --> - --> 1 --> =, updates display with expected calculated value', ()=>{
			const expectedResult = '4';
			
			wrapper.find('button[children="5"]').simulate('click');			
			wrapper.find('button[children="-"]').simulate('click');
			wrapper.find('button[children="1"]').simulate('click');
			wrapper.find('button[children="="]').simulate('click');

			expect(wrapper.find('.display').text()).toBe(expectedResult);
		});

		it('click --> 1  --> - --> 5 --> =, updates display with expected calculated value', ()=>{
			const expectedResult = '-4';
			
			wrapper.find('button[children="1"]').simulate('click');			
			wrapper.find('button[children="-"]').simulate('click');
			wrapper.find('button[children="5"]').simulate('click');
			wrapper.find('button[children="="]').simulate('click');

			expect(wrapper.find('.display').text()).toBe(expectedResult);
		});


		it('click --> 7 --> 0 --> %, updates display with expected calculated value', ()=>{
			const expectedResult = '0.7';

			wrapper.find('button[children="7"]').simulate('click');
			wrapper.find('button[children="0"]').simulate('click');			
			wrapper.find('button[children="%"]').simulate('click');
			
			expect(wrapper.find('.display').text()).toBe(expectedResult);
		});

		it('click --> 1 --> 5 -- > + --> 5 --> + --> 6 --> +, updates display with expected calculated value', ()=>{
			const expectedResult = '26';
			
			wrapper.find('button[children="1"]').simulate('click');
			wrapper.find('button[children="5"]').simulate('click');
			wrapper.find('button[children="+"]').simulate('click');
			wrapper.find('button[children="5"]').simulate('click');
			wrapper.find('button[children="+"]').simulate('click');
			wrapper.find('button[children="6"]').simulate('click');
			wrapper.find('button[children="+"]').simulate('click');

			expect(wrapper.find('.display').text()).toBe(expectedResult);
		});

		it('click --> 5 --> + --> 5 --> X --> 6 =, updates display with expected calculated value', ()=>{
			const expectedResult = '35';
			
			wrapper.find('button[children="5"]').simulate('click');
			wrapper.find('button[children="+"]').simulate('click');
			wrapper.find('button[children="5"]').simulate('click');
			wrapper.find('button[children="X"]').simulate('click');
			wrapper.find('button[children="6"]').simulate('click');
			wrapper.find('button[children="="]').simulate('click');
			
			expect(wrapper.find('.display').text()).toBe(expectedResult);
		});

		it('click --> 5 --> - --> 5 --> X --> 6 =, updates display with expected calculated value', ()=>{
			const expectedResult = '-25';
			
			wrapper.find('button[children="5"]').simulate('click');
			wrapper.find('button[children="-"]').simulate('click');
			wrapper.find('button[children="5"]').simulate('click');
			wrapper.find('button[children="X"]').simulate('click');
			wrapper.find('button[children="6"]').simulate('click');
			wrapper.find('button[children="="]').simulate('click');
			
			expect(wrapper.find('.display').text()).toBe(expectedResult);
		});

		it('click --> 5 --> + --> 30 --> ÷ --> 6 =, updates display with expected calculated value', ()=>{
			const expectedResult = '10';
			
			wrapper.find('button[children="5"]').simulate('click');
			wrapper.find('button[children="+"]').simulate('click');
			wrapper.find('button[children="3"]').simulate('click');
			wrapper.find('button[children="0"]').simulate('click');
			wrapper.find('button[children="÷"]').simulate('click');
			wrapper.find('button[children="6"]').simulate('click');
			wrapper.find('button[children="="]').simulate('click');
			
			expect(wrapper.find('.display').text()).toBe(expectedResult);
		});

		it('click --> 5 --> * --> 5 --> =, updates display with expected calculated value', ()=>{
			const expectedResult = '25';
			
			wrapper.find('button[children="5"]').simulate('click');
			wrapper.find('button[children="X"]').simulate('click');
			wrapper.find('button[children="5"]').simulate('click');
			wrapper.find('button[children="="]').simulate('click');
			
			expect(wrapper.find('.display').text()).toBe(expectedResult);
		});

		it('click --> 4 --> ÷ --> 2 --> =, updates display with expected calculated value', ()=>{
			const expectedResult = '2';
			
			wrapper.find('button[children="4"]').simulate('click');
			wrapper.find('button[children="÷"]').simulate('click');
			wrapper.find('button[children="2"]').simulate('click');
			wrapper.find('button[children="="]').simulate('click');
			
			expect(wrapper.find('.display').text()).toBe(expectedResult);
		});
		
		it('click --> 2 --> ÷ --> 4 --> =, updates display with expected calculated value', ()=>{
			const expectedResult = '0.5';
			
			wrapper.find('button[children="2"]').simulate('click');
			wrapper.find('button[children="÷"]').simulate('click');
			wrapper.find('button[children="4"]').simulate('click');
			wrapper.find('button[children="="]').simulate('click');
			
			expect(wrapper.find('.display').text()).toBe(expectedResult);
		});

		it('click --> 2 --> ÷ --> 0 --> =, shows error in display', ()=>{		
			
			wrapper.find('button[children="2"]').simulate('click');
			wrapper.find('button[children="÷"]').simulate('click');
			wrapper.find('button[children="0"]').simulate('click');
			wrapper.find('button[children="="]').simulate('click');
			
			expect(wrapper.find('.display').text()).toContain('Cannot divide by Zero');
		});

		it('click --> 2 --> + --> 1 --> 2 --> =, shows error in display', ()=>{		
			const expectedResult = '14';
			wrapper.find('button[children="2"]').simulate('click');
			wrapper.find('button[children="+"]').simulate('click');
			wrapper.find('button[children="1"]').simulate('click');
			wrapper.find('button[children="2"]').simulate('click');
			wrapper.find('button[children="="]').simulate('click');
			
			expect(wrapper.find('.display').text()).toBe(expectedResult);
		});

		it('click --> - --> - , keeps zero in display', ()=>{		
			const expectedResult = '0';
			wrapper.find('button[children="-"]').simulate('click');
			wrapper.find('button[children="-"]').simulate('click');
			
			expect(wrapper.find('.display').text()).toBe(expectedResult);
		});

		it('click --> 4 --> 5 --> + --> 3 --> = --> X --> 2 --> =, updates display with expected calculated value', ()=>{		
			const expectedResult = '24';
			wrapper.find('button[children="4"]').simulate('click');
			wrapper.find('button[children="5"]').simulate('click');
			wrapper.find('button[children="+"]').simulate('click');
			wrapper.find('button[children="3"]').simulate('click');
			wrapper.find('button[children="="]').simulate('click');
			wrapper.find('button[children="÷"]').simulate('click');
			wrapper.find('button[children="2"]').simulate('click');
			wrapper.find('button[children="="]').simulate('click');
			
			expect(wrapper.find('.display').text()).toBe(expectedResult);
		});
	});
});