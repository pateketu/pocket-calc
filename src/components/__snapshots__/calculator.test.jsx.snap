// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders as expeced 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Calculator>
    <div
      className="calculator"
    >
      <Connect(Display)>
        <Display
          dispatch={[Function]}
          value={0}
        >
          <div
            className="display"
          >
            0
          </div>
        </Display>
      </Connect(Display)>
      <Connect(KeyPad)>
        <KeyPad
          clear={[Function]}
          currentOperation="UNKNOWN"
          error={[Function]}
          input={[Function]}
          operation={[Function]}
          point={[Function]}
        >
          <div
            className="keyPad"
          >
            <div
              className="operationKeys"
            >
              <Key
                onClick={[Function]}
                value="C"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  C
                </button>
              </Key>
              <Key
                onClick={[Function]}
                value="+/-"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  +/-
                </button>
              </Key>
              <Key
                onClick={[Function]}
                value="%"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  %
                </button>
              </Key>
            </div>
            <div
              className="numberKeys"
            >
              <Key
                onClick={[Function]}
                value="0"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  0
                </button>
              </Key>
              <Key
                onClick={[Function]}
                value="."
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  .
                </button>
              </Key>
              <Key
                onClick={[Function]}
                value="1"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  1
                </button>
              </Key>
              <Key
                onClick={[Function]}
                value="2"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  2
                </button>
              </Key>
              <Key
                onClick={[Function]}
                value="3"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  3
                </button>
              </Key>
              <Key
                onClick={[Function]}
                value="4"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  4
                </button>
              </Key>
              <Key
                onClick={[Function]}
                value="5"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  5
                </button>
              </Key>
              <Key
                onClick={[Function]}
                value="6"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  6
                </button>
              </Key>
              <Key
                onClick={[Function]}
                value="7"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  7
                </button>
              </Key>
              <Key
                onClick={[Function]}
                value="8"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  8
                </button>
              </Key>
              <Key
                onClick={[Function]}
                value="9"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  9
                </button>
              </Key>
            </div>
            <div
              className="arthmaticOperationKeys"
            >
              <Key
                isActive={false}
                onClick={[Function]}
                value="รท"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  รท
                </button>
              </Key>
              <Key
                isActive={false}
                onClick={[Function]}
                value="X"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  X
                </button>
              </Key>
              <Key
                isActive={false}
                onClick={[Function]}
                value="-"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  -
                </button>
              </Key>
              <Key
                isActive={false}
                onClick={[Function]}
                value="+"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  +
                </button>
              </Key>
              <Key
                isActive={false}
                onClick={[Function]}
                value="="
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  =
                </button>
              </Key>
            </div>
          </div>
        </KeyPad>
      </Connect(KeyPad)>
    </div>
  </Calculator>
</Provider>
`;
